core/blockchain.go

```diff
@@ -1726,6 +1748,7 @@ func (bc *BlockChain) insertChain(chain types.Blocks, setHead bool) (int, error)
 		if err != nil {
 			return it.index, err
 		}
+		statedb.SetLogger(bc.logger)
```

Comment:
(Comment unrelated to line, comment for line 1707 above)

At https://github.com/ethereum/go-ethereum/blob/3cabef4c66575f641858ff5ce63120288f1cab44/core/blockchain.go#L1707-L1715

We skip blocks that are already known. Over the last three years, it happened to us to hit this condition while syncing some Ethereum chains (so including those using a fork of `geth`). Our current instrumentation https://github.com/streamingfast/go-ethereum/blob/firehose-v2/core/blockchain.go#L1697C12-L1705 does instrument this path.

We could argue that the block should already have been seen so it's ok to skip it. But sometimes node operators bootstrap themselves from pre-existing data dir and the condition could be triggered there.

I think it's safer to log those nonetheless and document the tracer of the low possibility to log the same block twice, behavior that could happen one some other occasions like on a chain rewind.

Reply:
There's a very slim chance this condition could happen on mainnet post-merge. I'd rather avoid double-emitting in this case.

Reply:
> There's a very slim chance this condition could happen on mainnet post-merge

Slim means there is a chance it happens.

> I'd rather avoid double-emitting in this case

From our post experience, there is no emitting at all happening. You used `double-emitting`, does it imply it is emitted as part of the current "process"?